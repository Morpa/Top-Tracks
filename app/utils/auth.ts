import dotenv from 'dotenv'

dotenv.config()

const state = {
  token: null,
  expires: 0
}

export const getToken = async () => {
  const now = Math.round(new Date().getTime() / 1000)
  const remaining = state.expires - now

  if (remaining <= 60) {
    state.token = null
    state.expires = 0
  }

  if (state.expires === 0) {
    const { access_token, expires_in } = await fetch(
      'https://accounts.spotify.com/api/token',
      {
        method: 'POST',
        headers: {
          Authorization: `Basic ${Buffer.from(
            process.env.SPOTIFY_CLIENT_ID +
              ':' +
              process.env.SPOTIFY_CLIENT_SECRET
          ).toString('base64')}`,
          'Content-Type': 'application/x-www-form-urlencoded'
        },
        body: 'grant_type=client_credentials'
      }
    ).then((response) => response.json())

    state.token = access_token
    state.expires = now + expires_in
  }

  return state.token
}
