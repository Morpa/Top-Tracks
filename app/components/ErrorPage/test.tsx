import { render, screen } from '~/utils/test-utils'
import { ErrorPage } from '.'

describe('<ErrorPage />', () => {
  it('should render the heading', () => {
    const { container } = render(<ErrorPage />)

    expect(
      screen.getByRole('img', { name: /Um robo quebrado/i })
    ).toBeInTheDocument()

    expect(
      screen.getByRole('heading', {
        name: /Algo deu errado. Tente novamente mais tarde!/i
      })
    ).toBeInTheDocument()

    expect(container.firstChild).toMatchSnapshot()
  })
})
