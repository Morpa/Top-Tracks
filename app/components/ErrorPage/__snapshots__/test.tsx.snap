// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ErrorPage /> should render the heading 1`] = `
.c3 {
  color: #FAFAFA;
  position: relative;
  margin-bottom: 3.2rem;
  font-size: 2.0rem;
}

.c3::after {
  position: absolute;
  left: 0;
  bottom: -1rem;
  content: '';
  width: 5rem;
  border-bottom: 0.5rem solid #3CD3C1;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  padding: 5.6rem;
}

.c1 {
  margin: 0.8rem;
}

@media (min-width:768px) {
  .c3 {
    font-size: 2.8rem;
  }
}

@media (max-width:768px) {
  .c0 .c2 {
    font-size: 1.6rem;
  }
}

@media (max-width:768px) {
  .c1 {
    width: 90%;
  }
}

<main
  class="c0"
>
  <img
    alt="Um robo quebrado"
    class="c1"
    src="/img/error.png"
  />
  <h2
    class="c2 c3"
    color="white"
  >
    Algo deu errado. Tente novamente mais tarde!
  </h2>
</main>
`;
